core/trino-main/src/main/java/io/trino/operator/MergeProcessorOperator.java

```diff
@@ -40,13 +41,20 @@ public static OperatorFactory createOperatorFactory(
             int rowIdChannel,
             int mergeRowChannel,
             List<Integer> redistributionColumns,
-            List<Integer> dataColumnChannels)
+            List<Integer> dataColumnChannels,
+            Set<Integer> nonNullColumnChannels)
     {
-        MergeRowChangeProcessor rowChangeProcessor = createRowChangeProcessor(merge, rowIdChannel, mergeRowChannel, redistributionColumns, dataColumnChannels);
+        MergeRowChangeProcessor rowChangeProcessor = createRowChangeProcessor(merge, rowIdChannel, mergeRowChannel, redistributionColumns, dataColumnChannels, nonNullColumnChannels);
         return createAdapterOperatorFactory(new Factory(operatorId, planNodeId, rowChangeProcessor));
     }
 
-    private static MergeRowChangeProcessor createRowChangeProcessor(MergeParadigmAndTypes merge, int rowIdChannel, int mergeRowChannel, List<Integer> redistributionColumnChannels, List<Integer> dataColumnChannels)
+    private static MergeRowChangeProcessor createRowChangeProcessor(
```

Comment:
Now that we have a switch expression, we could inline this function

Reply:
Inlined.